File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/framework/api/apifile/FileApi.java
package ru.usedesk.chat_sdk.internal.data.framework.api.apifile;

import androidx.annotation.NonNull;

import java.util.List;

import javax.inject.Inject;

import okhttp3.MultipartBody;
import retrofit2.Response;
import ru.usedesk.chat_sdk.internal.data.framework.api.apifile.entity.FileResponse;
import ru.usedesk.chat_sdk.internal.data.framework.httpapi.IHttpApi;
import ru.usedesk.common_sdk.external.entity.exceptions.UsedeskHttpException;
import ru.usedesk.common_sdk.internal.api.IUsedeskApiFactory;

public class FileApi implements IFileApi {

    private final IUsedeskApiFactory apiFactory;

    @Inject
    public FileApi(@NonNull IUsedeskApiFactory apiFactory) {
        this.apiFactory = apiFactory;
    }

    @Override
    @NonNull
    public FileResponse post(@NonNull String baseUrl,
                             @NonNull List<MultipartBody.Part> request) throws Exception {
        IHttpApi httpApi = apiFactory.getInstance(baseUrl, IHttpApi.class);
        Response<FileResponse> response = httpApi.postFile(request).execute();
        if (response.isSuccessful() && response.code() == 200) {
            return response.body();
        }
        throw new UsedeskHttpException("Server error: " + response.code());
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/data/repository/api/loader/apifile/FileApi.kt
package ru.usedesk.chat_sdk.data.repository.api.loader.apifile

import okhttp3.MultipartBody
import ru.usedesk.chat_sdk.data.repository._extra.retrofit.IHttpApi
import ru.usedesk.chat_sdk.data.repository.api.loader.apifile.entity.FileResponse
import ru.usedesk.common_sdk.api.IUsedeskHttpApiFactory
import toothpick.InjectConstructor
import java.io.IOException

@InjectConstructor
internal class FileApi(
        private val httpApiFactory: IUsedeskHttpApiFactory
) : IFileApi {

    override fun post(baseUrl: String, request: List<MultipartBody.Part>): FileResponse {
        val httpApi = httpApiFactory.getInstance(baseUrl, IHttpApi::class.java)
        val response = httpApi.postFile(request).execute()
        if (response.isSuccessful) {
            response.body()?.also {
                return it
            }
        }
        throw IOException("Server error: " + response.code())
    }
}